### PR TITLE
[Enhancement] support refresh mv in reversed order

### PR DESCRIPTION
**Pull request by murphyatwork**
Fixes #issue

1. Support refresh partitions in reversed order, which is more reasonable when there're a lot of partitions
2. The default behavior has not been changed, but could be changed by `admin set frontend config('default_mv_partition_refresh_reverse'='true')`

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by satanson — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMvRefreshProcessor.java [hunk 1]
```diff
@@ -360,39 +359,48 @@ public void filterPartitionByRefreshNumber(Set<String> partitionsToRefresh, Mate
         if (partitionRefreshNumber <= 0) {
```
Comment (LiShuMing) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMvRefreshProcessor.java:359]: Do we need to consider getAutoRefreshPartitionsLimit in `getPartitionNamesByRangeWithPartitionLimit` too?

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMvRefreshProcessor.java [hunk 2]
```diff
@@ -360,39 +359,48 @@ public void filterPartitionByRefreshNumber(Set<String> partitionsToRefresh, Mate
...
-        String nextPartitionStart = null;
-        String endPartitionName = null;
-        if (partitionNameIter.hasNext()) {
-            String startPartitionName = partitionNameIter.next();
-            Range<PartitionKey> partitionKeyRange = mappedPartitionsToRefresh.get(startPartitionName);
-            nextPartitionStart = AnalyzerUtils.parseLiteralExprToDateString(partitionKeyRange.lowerEndpoint(), 0);
-            endPartitionName = startPartitionName;
-            partitionsToRefresh.remove(endPartitionName);
+        // Skip first N partitions, which will be refreshed in this round
+        List<String> residualPartitions = sortedPartition.subList(partitionRefreshNumber, sortedPartition.size());
```
[the remark this answers is not in the document]
Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMvRefreshProcessor.java]: fixed

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMvRefreshProcessor.java (hunk 1 above)
Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMvRefreshProcessor.java:359]: it's already considered ? I don't understand

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 3]
```diff
@@ -2546,6 +2546,10 @@ public class Config extends ConfigBase {
     @ConfField(mutable = true)
     public static long mv_active_checker_interval_seconds = 60;
 
+    // TODO: change the default value to true
+    @ConfField(mutable = true)
+    public static boolean default_mv_partition_refresh_reverse = false;
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:2551]: why set configuration? session variable is preferred

**Review by LiShuMing — APPROVED** [review]

**Pull request closed without merging**